metaflow/sidecar.py

```diff
@@ -48,109 +49,184 @@ def poll(self, timeout):
 
 
 class SidecarSubProcess(object):
-    def __init__(self, worker_type):
-        # type: (str) -> None
-        self.__worker_type = worker_type
-        self.__process = None
-        self.__poller = None
+    def __init__(self, worker_type, context=None):
+        # type: (str, dict) -> None
+        self._worker_type = worker_type
+        self._process = None
+        self._poller = None
+        self._context = context
+        self._send_context_remaining_tries = (
+            CONTEXT_RETRY_TIMES if self._context is not None else 0
```

Comment:
This seems highly specific to the logger use case here.

Reply:
How so? It basically ensures that the context is sent prior to sending a message. Or at least gives it a real go at sending it.